### ISSUE TITLE
Support GetPublicPath from federation options

### ISSUE THREAD

**Issue by judithhartmann** [module-federation/vite#104]
Currently setting `getPublicPath` in the plugin config seems to have no effect (I could not find the function i passed there anywhere in my build output).

Expected behaviour described here:
https://module-federation.io/configure/getpublicpath.html

**Comment by zhangHongEn** [comment]
understand

**Comment by gioboa** [comment]
@judithhartmann we are working on it. can you create an example based on [this example](https://github.com/module-federation/vite/tree/main/examples/vite-webpack-rspack) of these special config pls?
You can post here the changes I guess. Thanks, this will help us a lot.

**Comment by judithhartmann** [comment]
I will.

I also saw in your PR that there seems to be a static public path option that i did not find in the docs, I will try that as well.

The issue i ran into my project is that we have the modules in a cdn but the remoteEntry was trying to load the `get` from a relative path, that of course did not match the actual location of that filw

**Comment by gioboa** [comment]
Thanks, with your example we can certify the new implementation 👏

**Comment by judithhartmann** [comment]
I tried to build an example from your PR #108, unfortunately i got stuck because getPublicPath does not seem to be applied.

I would assume i'll find something related to it in the build output of the remote.
i basically just set  `getPublicPath: `function () { return 'http://localhost:4001/custom-prefix/'}`,` in the remote, but i could not find '/custom-prefix' in the output at all.
if it works i would find some traces of it right?

**Comment by gioboa** [comment]
Yep, the PR it's not complete yet. I was waiting for the test case. Can you share with me your configurations please ?

**Comment by judithhartmann** [comment]
Here are all my modifications
https://github.com/judithhartmann/module-federation-vite/commit/01502f5c86f1aba00d09675e165642c5bade92b7

I modified the vite dev and preview server, so they will serve all files with the `/custom-prefix` as well, without using `base`.

Basically the test would be, that every file requested from the host also contains the custom prefix.

**Comment by gioboa** [comment]
@judithhartmann I'm wondering if we can consider using the Vite [experimental.renderBuiltUrl](https://vitejs.dev/guide/build#advanced-base-options) configuration instead of supporting the `getPublicPath` option.

**Comment by judithhartmann** [comment]
it looks like this might work

however, from user perspective, since since plugin does not have it's own extensive docs, but relies on https://module-federation.io/ it might be confusing, as getPublicPath is documented over there to achieve this behaviour

**Comment by gioboa** [comment]
Yup, you are right, we can add a section in the README for that. Would you like to drop a PR for that?

**Comment by gioboa** [comment]
I added a [warning](https://github.com/module-federation/vite/pull/108#issue-2548901482) 
This is nice for the DX. @judithhartmann Can you verify that that Vite config is solving the case pls?

**Comment by zhangHongEn** [comment]
Even if `[experimental.renderBuiltUrl](https://vitejs.dev/guide/build#advanced-base-options)` is set, manifest.json still needs to have `getPublicPath`, and they need to work together. I’ve restored the `36f1e2e1e571fa8a3f62791d45f44b67f6d4c093` code.

**Comment by judithhartmann** [comment]
Hi @gioboa, sorry for the delay.

I tried out the experimental.renderBuiltUrl. ([Branch](https://github.com/judithhartmann/module-federation-vite/tree/feat/getPublicPath-example))

```js
  experimental: {
    renderBuiltUrl: (filename) => {
      return `http://localhost:4001/custom-prefix/${filename}`;
    },
  },
```

Unfortunately the only place where i can find the `custom-prefix` in the built output is in the index.html, so it does not have an influence on the remotyEntry.js nor any files related to it.

In the vite dev example, all files aside from the remoteEntry.js where loaded without the prefix 

combined with @zhangHongEn concerns, i think we need to reopen this.

**Comment by artuska** [comment]
Any success on this?

Got

```
Uncaught SyntaxError: Cannot use import statement outside a module (at remoteEntry.js:1:1)
```

because remoteEntry.js:1:1 is importing assets from `./`:

```
import { i as index_cjs, a as iplatform_client_footer__mf_v__runtimeInit__mf_v__ } from './iplatform_client_footer__mf_v__runtimeInit__mf_v__-DvBgc6E1.js';
```

Setting

```
getPublicPath: `function() {return "/foo_bar_prefix"}`,
```

in Vite module-federation config just do nothing.
